core/controllers/access_validators.py

```diff
@@ -0,0 +1,181 @@
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controllers for the validating access."""
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from constants import constants
+from core.controllers import acl_decorators
+from core.controllers import base
+from core.domain import classroom_services
+from core.domain import user_services
+
+import feconf
+
+from typing import Any # pylint: disable=unused-import
+from typing import Dict # pylint: disable=unused-import
+from typing import Text # pylint: disable=unused-import
+
+
+class SplashAccessValidationHandler(base.BaseHandler):
```

Comment:
@ashutoshc8101 Can you please create an issue and add a TODO comment in case we are going to follow the finalized structure later? Also, please leave a comment for the reviewer explaining the issue with this structure and what the final structure!

Reply:
let me first discuss the approach with srijan and sean. Once, they approve it, I will leave a comment.

Reply:
@ashutoshc8101 What is the resolution on this comment?

Reply:
done, created issue #13605.
During the discussion about the approach with Sean, he raised a concern about how schema validation will be done if URLs for different validation handlers can have different URL fragments. I have raised this question in the doc and have assigned the comment to you. Let's discuss this in the doc and not block this PR for now.